main.go

```diff
@@ -91,8 +91,6 @@ func main() {
 		r.Use(cors.New(corsConfig))
 	}
 
-	HealthModule{}.Register(r)
-
 	setupBasicAuth(r, config)
 
```

Comment:
I think this was deliberately like this so that the basic auth was not applied to the HealthModule's endpoints. Try setting the auth environment variable and see if you can still access these endpoints anonymously, or if you need to revert the changes in this file.

If you want to make them require auth as well, I suggest you do that in a different PR as we need to make sure that our helm chart will properly provide Kubernetes with the right credentials when it's doing the liveness/readiness probes.

IMO leaving them as anonymous just simplifies so much.

---

One of the quirks with how we have it set up now is that you can only access the `/` and `/health` endpoints if you're inside the network, such as how Kubernetes is, while all users will not be given access as we have reverse proxies filtering out everything not starting with `/api/`. So we don't even need auth for those endpoints as they are not accessible to the "public" anyway.

Perhaps we could change the swagger documentation instead, and leave the endpoints as-is?

Reply:
A lot more behind this way of doing it than I expected.

It seems like changing the swagger documentation instead would be simpler and, dare I say, just as effective?
In that case I think that would be the best way to do it. I don't know in what way to go about changing the swagger docs though. If I get pointed in the right direction then I can do it, if nobody else wants to do it that is.

Reply:
Hmm so if we change the swagger docs then you still wont be able to access it as the reverse proxy will point the requests elsewhere (probably towards the frontend). Changing the endpoints here does help with that.

What I propose at least is to keep the old endpoints for backward compatibility (look I spelled it right this time!) and move the endpoints as you have. So they're accessible from both places, and in next major release or something we can remove the old endpoints and perhaps replace the `/` endpoint with something else, like just a simple "hello from wharf-api" or something.

And then also add in the CHANGELOG.md a "Deprecated" note about the `/` and `/health` endpoints being moved soon to `/api/health` and `/api/ping`

Reply:
Pushed new commits.

> backward compatibility (look I spelled it right this time!)

Wooh!